src/main/java/tan/TaskManager.java

```diff
@@ -0,0 +1,223 @@
+package tan;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class TaskManager {
+    private static List<Task> listOfTasks = new LinkedList<Task>();
+    private static int totalNumberOfTask = 0;
+
+    public TaskManager() {
+
+    }
+
+    /**
+     * Returns the Task which is at
+     * the index, null otherwise.
+     *
+     * @param indexTask index of the task to obtain.
+     * @return Task at specified index.
+     */
+    private static Task getTask(int indexTask) {
+        if (indexTask > listOfTasks.size() || indexTask < 0) {
+            System.out.println("Task is out of range!");
+            return null;
+        }
+        return listOfTasks.get(indexTask);
+    }
+
+    /**
+     * Marks a certain task as done.
+     *
+     * @param indexTask The index of the task to be marked as done.
+     */
+    public static void markTaskAsDone(int indexTask) {
+        if (indexTask > listOfTasks.size() || indexTask < 0) {
+            System.out.println("No such task!");
+            return;
+        }
+        Task currentTask = getTask(indexTask - 1);
+        if (currentTask != null) {
+            currentTask.markAsDone();
+        }
+    }
+
+    /**
+     * Takes the user input as a string
+     * and decides what type of command it is and
+     * executes its respective jobs.
+     * If none of the inputs matches any of the command,
+     * It tells the user to try again.
+     *
+     * @param userInput The entire input from the user.
+     */
+    public static void addTask(String userInput) {
+        String typeOfTask = getTypeOfTask(userInput);
+        Task curTask = null;
+        switch (typeOfTask) {
+        case "todo":
+            String todoDesc = getDescriptionOfToDo(userInput);
+            if (todoDesc == null) {
+                //if user did not key in any description.
+                System.out.println("Please enter a valid description!");
+                return;
+            }
+            curTask = new ToDo(todoDesc);
+            listOfTasks.add(curTask);
+            break;
+        case "deadline":
+            String deadlineDesc = getDescriptionOfDeadline(userInput);
+            String deadlineDateTime = getDateTimeOfDeadline(userInput);
+            curTask = new Deadline(deadlineDesc, deadlineDateTime);
+            listOfTasks.add(curTask);
```

Comment:
Perhaps you could extract out common code between all your case blocks to simplify your code